cmd/cluster-agent/subcommands/start/command.go

```diff
@@ -350,6 +332,29 @@ func start(log log.Component,
 		pkglog.Warn("Failed to auto-detect a Kubernetes cluster name. We recommend you set it manually via the cluster_name config option")
 	}
 
+	// Initialize and start remote configuration client
+	var rcClient *rcclient.Client
+	rcserv, isSet := rcService.Get()
+	if pkgconfig.IsRemoteConfigEnabled(config) && isSet {
+		// TODO: Is APM Tracing always required? I am not sure
+		products := []string{state.ProductAPMTracing}
+
+		if config.GetBool("autoscaling.workload.enabled") {
+			products = append(products, state.ProductContainerAutoscalingSettings, state.ProductContainerAutoscalingValues)
+		}
+
+		var err error
+		rcClient, err = initializeRemoteConfigClient(rcserv, config, clusterName, clusterID, products...)
```

Comment:
Asking because I don't have enough context. Are we fine with moving the remoteConfigClient down here ? If the remoteConfigClient fetches some config on startup, we might have a different behavior.

Also I think we should try to remove the config pkg but it can be done later.

Reply:
IMO the RC client can be implemented as a component as well but it's not for us either.